### PR TITLE
Re-implementation of ATLAS single top

### PR DESCRIPTION
**Pull request by jacoterh**
I am opening this PR as draft to make discussion easier - it is not ready to merge yet. I have a question regarding the old implementation of the systematics in the ATLAS_SINGLETOP_7TEV_T-Y-NORM dataset. 

Looking at the [legacy implementation](https://github.com/NNPDF/nnpdf/blob/b16abc59408ff290e8aff65a3aa4d78d538d04d0/nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_7TEV/uncertainties_legacy_T-Y-NORM.yaml), I find 17 systematics per datapoint while the experimental paper only reports 7, see Table XV in [[1406.7844]](https://arxiv.org/pdf/1406.7844). Now, 3 of these correspond to correlated statistical uncertainties, which leaves 14 systematics, which is **twice** the number of real systematics! I went back to the[ old common data implementation](https://github.com/NNPDF/nnpdf/blob/a1965caf453d8be1b00529e4a19f7c45ea3483bf/nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_7TEV/old_commondata_script.cc) (the c++ implementation) and found that both the upper and lower bounds are stored, e.g. relevant in case of asymmetric uncertainties.  See line 208 in the c++ script.

However, my understanding was that symmetrising was performed before writing the commondata! So I am not sure why should keep both bounds for each source.

### CODE REVIEW

**Comment by scarlehoff** [comment]
_if_ you get the same covmat (and t0 covmat) synmetrizing I'd say we can keep the symmetrized version and drop the other

but @enocera might know better

**Comment by jacoterh** [comment]
The implementation is done except for the theory/data comparison and x-Q2 plots. The covariance matrices were compared with 

```
import numpy as np
from validphys.api import API

observables = ["T-Y-NORM", "TBAR-Y-NORM", "TCHANNEL-XSEC"]

for obs in observables:

    new_implementation = f"ATLAS_SINGLETOP_7TEV_{obs}"
    old_implementation = f"ATLAS_SINGLETOP_7TEV_{obs}"

    inp1 = {
        "dataset_input": {"dataset": f"{new_implementation}"},
        "theoryid": 40_000_000,
        "use_cuts": "internal",
        "t0pdfset": "NNPDF40_nnlo_as_01180",
        "use_t0": True,
    }
    inp2 = {
        "dataset_input": {"dataset": f"{old_implementation}", "variant": "legacy"},
        "theoryid": 40_000_000,
        "use_cuts": "internal",
        "t0pdfset": "NNPDF40_nnlo_as_01180",
        "use_t0": True,
    }

    covmat1 = API.covmat_from_systematics(**inp1)
    covmat2 = API.covmat_from_systematics(**inp2)

    t0_covmat1 = API.t0_covmat_from_systematics(**inp1)
    t0_covmat2 = API.t0_covmat_from_systematics(**inp2)

    print(f"Comparison for {new_implementation}")
    print(np.all(np.isclose(covmat1, covmat2)))
    print(np.all(np.isclose(t0_covmat1, t0_covmat2)))

```

with output

```
Comparison for ATLAS_SINGLETOP_7TEV_T-Y-NORM
False
False
Comparison for ATLAS_SINGLETOP_7TEV_TBAR-Y-NORM
False
False
Comparison for ATLAS_SINGLETOP_7TEV_TCHANNEL-XSEC
False
False
```

The reasons are as follows:

- `ATLAS_SINGLETOP_7TEV_T-Y-NORM`:  treatment of asymmetric uncertainties differs from the legacy implementation as discussed above.
- `ATLAS_SINGLETOP_7TEV_TBAR-Y-NORM`:  dito
- `ATLAS_SINGLETOP_7TEV_TCHANNEL-XSEC`: The legacy implementation discards systematics below 1%. Uncertainties contributing less than 1.0 %  are marked with "" in the paper, but on HEPdata these uncertainties are approximated with 0.5% . I implemented also these. The ones that contribute more than 1% agree with the legacy implementation.


Some questions that came up:
- Do we multiply relative uncertainties by the shifted central value or the unshifted one? See inline comment.
- I'm really not sure how to find out the correct type of uncertainties for some of the systematics. The luminositiy is clearly multiplicative, but for e.g. lepton uncertainties, scale variations, Parton shower, etc... I'm not at all sure.

File: nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_7TEV/metadata.yaml
Comment (scarlehoff) [reply nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_7TEV/metadata.yaml:84]: With respect to k1/k2/k3, this comment I left in #2178 applies here as well: https://github.com/NNPDF/nnpdf/pull/2178#pullrequestreview-2409495875 basically, look in process option and use one of the process there. Since (for single top) there will be no process I believe, you need to create it.

That means that you also need to add a function to define the xq coverage plot (and you are free to use as input variables what you want as long as you get the x-q plot as the output).

The best way to check is to do a data-theory comparison plot and a x-q coverage plot `{@plot_xq2@}` in master and this branch and seeing that they look the same.

**Comment by jacoterh** [comment]
Following up on @scarlehoff suggestion in [PR#2185](https://github.com/NNPDF/nnpdf/pull/2185) to compare the overall experimental chi2 with all datasets combined, I'm attaching here the reports:
- [ATLAS_SINGLETOP_7TEV](https://vp.nnpdf.science/uwRld4cVR4eaXdKjZFY1kw==/)
- [ATLAS_SINGLETOP_8TEV](https://vp.nnpdf.science/h4U3BMPCSRqJ2sOXA-luoQ==/)
- [ATLAS_SINGLETOP_13TEV](https://vp.nnpdf.science/5JUx2vbkRA6X3ys52oOAJg==/)

Regarding the ordering, legacy always appears first and is followed by the new commondata implementation, e.g 
```
- { dataset: ATLAS_SINGLETOP_8TEV_T-RAP-NORM, variant: legacy }
- { dataset: ATLAS_SINGLETOP_8TEV_T-RAP-NORM}
- { dataset: ATLAS_SINGLETOP_8TEV_TBAR-RAP-NORM, variant: legacy }
- { dataset: ATLAS_SINGLETOP_8TEV_TBAR-RAP-NORM}
- { dataset: ATLAS_SINGLETOP_8TEV_TCHANNEL-XSEC, variant: legacy }
- { dataset: ATLAS_SINGLETOP_8TEV_TCHANNEL-XSEC}
```

**Review by scarlehoff — APPROVED** [review]
Hi @jacoterh, looks good. The only thing, could you please bump the verison number and change the comment that currently says "port of old commondata", which happily is no longer true :)

RE the comparison you need to separate the legacy and the new (so that you can check whether the correlations between datasets are captured in the same way).

For reference, this is the runcard I'm using to test:

<details>
<summary> vp runcard </summary>

```
meta:
  title: Data vs Th
  keywords: comparison
  author: juacrumar

pdfs:
  - id: NNPDF40_nnlo_as_01180
    label: NNPDF4.0 NNLO
pdf: NNPDF40_nnlo_as_01180

#theoryid: 40_000_000
theoryid: 200

use_cuts: "internal"
marker_by: "dataset"

old_and_new:
  - temporal: "OLD data"
    dataset_inputs:
    - { dataset: ATLAS_SINGLETOP_7TEV_T-Y-NORM, variant: legacy }
    - { dataset: ATLAS_SINGLETOP_7TEV_TBAR-Y-NORM, variant: legacy }
    - { dataset: ATLAS_SINGLETOP_7TEV_TCHANNEL-XSEC, variant: legacy }
    - { dataset: ATLAS_SINGLETOP_8TEV_T-RAP-NORM, variant: legacy }
    - { dataset: ATLAS_SINGLETOP_8TEV_TBAR-RAP-NORM, variant: legacy }
    - { dataset: ATLAS_SINGLETOP_8TEV_TCHANNEL-XSEC, variant: legacy }
    - { dataset: ATLAS_SINGLETOP_13TEV_TCHANNEL-XSEC, variant: legacy }
  - temporal: "NEW data"
    dataset_inputs:
    - { dataset: ATLAS_SINGLETOP_7TEV_T-Y-NORM}
    - { dataset: ATLAS_SINGLETOP_7TEV_TBAR-Y-NORM}
    - { dataset: ATLAS_SINGLETOP_7TEV_TCHANNEL-XSEC}
    - { dataset: ATLAS_SINGLETOP_8TEV_T-RAP-NORM}
    - { dataset: ATLAS_SINGLETOP_8TEV_TBAR-RAP-NORM}
    - { dataset: ATLAS_SINGLETOP_8TEV_TCHANNEL-XSEC}
    - { dataset: ATLAS_SINGLETOP_13TEV_TCHANNEL-XSEC}

template_text: |
  {@ with old_and_new @}
  {@ temporal @}
  ==============

  Data-TH comparison
  ------------------
  {@ dataset_inputs plot_fancy @}

  Normalized
  ----------
  {@ dataset_inputs::pdfs plot_fancy(normalize_to=data)@}

  xq mapping
  ----------
  {@ plot_xq2 @}

  Chi Data
  --------
  {@ experiments_chi2_table @}
  {@ total_chi2_data @}
  {@ endwith @}

actions_:
  - report(main=true)
```

</details>

**Comment by scarlehoff** [comment]
Btw, before the merge, please rebase on top of master (you can do that from github by pressing the arrow where it says update branch and selecting update with rebase)

**Comment by scarlehoff** [comment]
RE the questions (but @enocera is the right person to answer)

> Do we multiply relative uncertainties by the shifted central value or the unshifted one? See inline comment.

I think we have decided to keep al uncertainties are multiplicative?

> I'm really not sure how to find out the correct type of uncertainties for some of the systematics. The luminositiy is clearly multiplicative, but for e.g. lepton uncertainties, scale variations, Parton shower, etc... I'm not at all sure.

See above.

> Which value of m_t, m_W, m_Z, etc.. should we use for the scales? I noticed this is not consistent for all datasets, i.e. sometimes m_t = 172.5, sometimes 173.3

It's not really important for the data since the scale has an effect only on the theory. Eventually this might come from the PineaAPPL grid as @felixhekhorn has been advocating for (since that's the right place, a theory might be using q = m_t and another one q = Et). So whatever the dataset says (if it says something) is ok.

**Comment by scarlehoff** [comment]
@jacoterh is this final now? Could you reabase (preferably) on top of master or merge master into this branch? (so that the tests and the bot that @Radonirinaunimi prepared can run on this data)

thanks

**Comment by RoyStegeman** [comment]
We're aware, it's the data with the capitalisation issue and macos isn't sensitive to capitalisation in filenames so should be resolved on a linux cluster.

**Comment by scarlehoff** [comment]
I see, I can fix it then.

I'll merge from `[aee1454](https://github.com/NNPDF/nnpdf/pull/2189/commits/aee145460b54fde870a7fc040d41862c588bd4d0)` if that's ok?

**Comment by RoyStegeman** [comment]
Up to @jacoterh 

I think we should also remove the commit that changes these EIC files so we don't get new changes to them in master and keep having to deal with it when rebasing

**Comment by scarlehoff** [comment]
If he only has a mac I don't think he can fix it?

I will squash and rebase, that should minimize the changes and make the rebase easy-ish.

**Comment by RoyStegeman** [comment]
He has a cluster 

> I will squash and rebase, that should minimize the changes and make the rebase easy-ish.

Thanks

**Comment by scarlehoff** [comment]
> He has a cluster

Then let me know what you prefer @jacoterh

File: nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_13TEV/metadata.yaml
Comment (scarlehoff) [inline nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_13TEV/metadata.yaml:59]:
```suggestion
```

Since the data hasn't changed, you can remove this file (and line)

File: nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_13TEV/metadata.yaml
Comment (scarlehoff) [inline nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_13TEV/metadata.yaml:32]: This is fine but not necessary, you can remove `sqrts` (which will be taken from the filename) and `zero` (we can just have one variable now and it is fine).

But it's fine, just if you want to save a few bytes (and it is really just a few for this dataset).

File: nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_13TEV/metadata.yaml
Comment (scarlehoff) [inline nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_13TEV/metadata.yaml:31]: If you want to put `m_t` feel free to make the change (up to you, you are the one that are going to be using this dataset the most I guess).

File: nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_7TEV/metadata.yaml
Comment (scarlehoff) [inline nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_7TEV/metadata.yaml:169]:
```suggestion
```

And the file can be removed as well.

File: nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_8TEV/metadata.yaml
Comment (scarlehoff) [inline nnpdf_data/nnpdf_data/commondata/ATLAS_SINGLETOP_8TEV/metadata.yaml:169]:
```suggestion
```

**Review by scarlehoff — APPROVED** [review]
Thanks. I just left a few comments about things that should be removed (repeated data files).

For the point about the kinematic variables (removing the unused ones and changing mt2 -> mt) do as you prefer.

**Pull request merged**
